### PR TITLE
Add new error to skip endpointslices with empty OwnerReferences

### PR DESCRIPTION
**Pull request by sabinaaledort**
Adding a new error to skip endpointslices with empty OwnerReferences instead of causing the entire matrix build to fail.

### CODE REVIEW

**Review by shirmoran — APPROVED** [review]

File: pkg/endpointslices/endpointslices.go [hunk 1]
```diff
@@ -105,9 +115,13 @@ func (ep *EndpointSlicesExporter) ToComDetails() ([]types.ComDetails, error) {
 	for _, epSliceInfo := range ep.sliceInfo {
 		cds, err := epSliceInfo.toComDetails(ep.nodeToRole)
 		if err != nil {
+			var noOwnerRefErr *NoOwnerRefErr
```
Comment (SchSeba) [inline pkg/endpointslices/endpointslices.go:118]: just a nit move this one outside the for loop

File: pkg/endpointslices/endpointslices.go [hunk 2]
```diff
@@ -29,6 +29,15 @@ type EndpointSlicesExporter struct {
 	sliceInfo  []EndpointSlicesInfo
 }
 
+type NoOwnerRefErr struct {
```
Comment (aabughosh) [inline pkg/endpointslices/endpointslices.go:32]: what do u think to add this to utils?
since its not EndpointSlice

Comment (SchSeba) [reply pkg/endpointslices/endpointslices.go:32]: I think we can leave it here as we don't use it in any other place and I don't want to make utils a kitchen sink

**Review by SchSeba — APPROVED** [review]
/lgtm
/approve

**Review by aabughosh — APPROVED** [review]

**Pull request merged**